libs/remix-ui/workspace/src/lib/actions/index.ts

```diff
@@ -0,0 +1,364 @@
+      } else {
+        // id is not existing, need to create a new gist
+        dispatch(displayPopUp('Creating a new gist ...'))
+        gists.create({
+          description: description,
+          public: true,
+          files: packaged
+        }, (error, result) => {
+          handleGistResponse(error, result)
+        })
+      }
+    }
+  } catch (error) {
+    console.log(error)
+    dispatch(displayNotification('Publish to gist Failed', 'Failed to create gist: ' + error.message, 'Close', null, async () => {}))
+  }
+}
+
+export const clearPopUp = async () => {
+  dispatch(hidePopUp())
+}
+
+export const createNewFile = async (path: string, rootDir: string) => {
+  const fileManager = plugin.fileManager
+  const newName = await createNonClashingNameAsync(path, fileManager)
+  const createFile = await fileManager.writeFile(newName, '')
+
+  if (!createFile) {
+    return dispatch(displayPopUp('Failed to create file ' + newName))
+  } else {
+    const path = newName.indexOf(rootDir + '/') === 0 ? newName.replace(rootDir + '/', '') : newName
+
+    await fileManager.open(path)
+    setFocusElement([{ key: path, type: 'file' }])
+  }
+}
+
+export const setFocusElement = async (elements: { key: string, type: 'file' | 'folder' | 'gist' }[]) => {
+  dispatch(focusElement(elements))
+}
+
+export const createNewFolder = async (path: string, rootDir: string) => {
+  const fileManager = plugin.fileManager
+  const dirName = path + '/'
+  const exists = await fileManager.exists(dirName)
+
+  if (exists) {
+    return dispatch(displayNotification('Rename File Failed', `A file or folder ${extractNameFromKey(path)} already exists at this location. Please choose a different name.`, 'Close', null, () => {}))
+  }
+  await fileManager.mkdir(dirName)
+  path = path.indexOf(rootDir + '/') === 0 ? path.replace(rootDir + '/', '') : path
+  dispatch(focusElement([{ key: path, type: 'folder' }]))
+}
+
+export const deletePath = async (path: string[]) => {
+  const fileManager = plugin.fileManager
+
+  for (const p of path) {
+    try {
+      await fileManager.remove(p)
+    } catch (e) {
+      const isDir = await fileManager.isDirectory(p)
+
+      dispatch(displayPopUp(`Failed to remove ${isDir ? 'folder' : 'file'} ${p}.`))
+    }
+  }
+}
+
+export const renamePath = async (oldPath: string, newPath: string) => {
+  const fileManager = plugin.fileManager
+  const exists = await fileManager.exists(newPath)
+
+  if (exists) {
+    dispatch(displayNotification('Rename File Failed', `A file or folder ${extractNameFromKey(newPath)} already exists at this location. Please choose a different name.`, 'Close', null, () => {}))
+  } else {
+    await fileManager.rename(oldPath, newPath)
+  }
+}
+
+export const copyFile = async (src: string, dest: string) => {
+  const fileManager = plugin.fileManager
+
+  try {
+    fileManager.copyFile(src, dest)
+  } catch (error) {
+    console.log('Oops! An error ocurred while performing copyFile operation.' + error)
```

Comment:
Is this needed to be in console also when it is on the popup? If yes, better to create a variable with that value and use at both places